modules/apigee/main.tf

```diff
@@ -0,0 +1,103 @@
+  project_id                           = var.project_id
+  authorized_network                   = var.organization.authorized_network
+  billing_type                         = var.organization.billing_type
+  runtime_type                         = var.organization.runtime_type
+  runtime_database_encryption_key_name = var.organization.database_encryption_key
+}
+
+resource "google_apigee_envgroup" "envgroups" {
+  for_each  = local.envgroups
+  name      = each.key
+  hostnames = each.value
+  org_id    = local.org_id
+}
+
+resource "google_apigee_environment" "environments" {
+  for_each        = local.environments
+  name            = each.key
+  display_name    = each.value.display_name
+  description     = each.value.description
+  deployment_type = each.value.deployment_type
+  api_proxy_type  = each.value.api_proxy_type
+  dynamic "node_config" {
+    for_each = try(each.value.node_config, null) != null ? [""] : []
+    content {
+      min_node_count               = node_config.min_node_count
+      max_node_count               = node_config.max_node_count
+      current_aggregate_node_count = node_config.current_aggregate_node_count
+    }
+  }
+  org_id = local.org_id
+}
+
+resource "google_apigee_envgroup_attachment" "envgroup_attachments" {
+  for_each = local.environments
+  envgroup_id = (lookup(local.envgroups, each.value.envgroup, null) == null ?
+  each.value.envgroup : google_apigee_envgroup.envgroups[each.value.envgroup].id)
+  environment = google_apigee_environment.environments[each.key].name
+}
+
+resource "google_apigee_environment_iam_binding" "binding" {
+  for_each = merge(concat([for k1, v1 in local.environments : {
+    for k2, v2 in coalesce(v1.iam, {}) : "${k1}-${k2}" => {
+      environment = "${k1}"
+      role        = k2
+      members     = v2
+    }
+  }])...)
+  org_id  = local.org_id
+  env_id  = google_apigee_environment.environments[each.value.environment].name
+  role    = each.value.role
+  members = each.value.members
+}
+
+resource "google_apigee_instance" "instances" {
+  for_each                 = local.instances
+  name                     = each.key
+  display_name             = each.value.display_name
+  description              = each.value.description
+  location                 = each.value.region
+  org_id                   = local.org_id
+  ip_range                 = each.value.psa_ip_cidr_range
+  disk_encryption_key_name = each.value.disk_encryption_key
+  consumer_accept_list     = each.value.consumer_accept_list
+}
+
+resource "google_apigee_instance_attachment" "instance_attachments" {
+  for_each = merge(concat([for k1, v1 in local.instances : {
+    for v2 in v1.environments :
+    "${k1}-${v2}" => {
+      instance    = k1
+      environment = v2
+    }
+  }])...)
+  instance_id = google_apigee_instance.instances[each.value.instance].id
+  environment = (lookup(local.environments, each.value.environment, null) == null ?
```

Comment:
try(local.environments[each.value.environment], "${local.org_id}/environments/${each.value.environment}")

Reply:
done, but it's not quite like you suggested